src/Mod/PartDesign/App/FeatureRevolution.cpp

```diff
@@ -183,7 +183,12 @@ App::DocumentObjectExecReturn *Revolution::execute(void)
 
 bool Revolution::suggestReversed(void)
 {
-    updateAxis();
+    try {
+        updateAxis();
+    } catch (const Base::Exception& e) {
+        return new App::DocumentObjectExecReturn(e.what());

```

Comment:
Are you sure the return type is correct here? It shoul dbe a boolean, so either "true" or "false". A pointer can be cast to a boolean by default (nullptr == false, everythign else == true)  and hence this compiles, but it is not what you actually want. This results in a memory leak, as the crewated pointer never gets free'd again, it just get lost in the cast.